gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -451,45 +457,48 @@ static boolean shouldIncludeMethodInGeneration(
                     + "Disregarding selective generation settings.",
                 librarySettingsList.get(0).getVersion(), protoPackage));
       }
-      return true;
+      return SelectiveGapicType.PUBLIC;
     }
     // librarySettingsList is technically a list, but is processed upstream and
     // only leave with 1 element. Otherwise, it is a misconfiguration and
     // should be caught upstream.
-    List<String> includeMethodsList =
-        librarySettingsList
-            .get(0)
-            .getJavaSettings()
-            .getCommon()
-            .getSelectiveGapicGeneration()
-            .getMethodsList();
-    // default to include all when nothing specified, this could be no java section
-    // specified in library setting, or the method list is empty
-    if (includeMethodsList.isEmpty()) {
-      return true;
-    }
+    SelectiveGapicGeneration selectiveGapicGenerationConfig =
+        librarySettingsList.get(0).getJavaSettings().getCommon().getSelectiveGapicGeneration();
+
+    List<String> includeMethodsList = selectiveGapicGenerationConfig.getMethodsList();
+
+    Boolean generateOmittedAsInternal =
+        selectiveGapicGenerationConfig.getGenerateOmittedAsInternal();
 
-    return includeMethodsList.contains(method.getFullName());
+    // Set method to PUBLIC if no SelectiveGapicGeneration Configuration is configured or the method
+    // is in the allow list.
+    // Otherwise, generate this method as INTERNAL or HIDDEN based on GenerateOmittedAsInternal
+    // flag.
+    if (includeMethodsList.isEmpty() && generateOmittedAsInternal == false
+        || includeMethodsList.contains(method.getFullName())) return SelectiveGapicType.PUBLIC;
+    else return generateOmittedAsInternal ? SelectiveGapicType.INTERNAL : SelectiveGapicType.HIDDEN;
```

Comment:
Perhaps simplify a bit this if-else statement for readability?
Also, as per go/java-style#s4.1.1-braces-always-used, can you add braces to the return statements?